mujoco_warp/_src/collision_primitive.py

```diff
@@ -232,6 +232,44 @@ def plane_box(
       break
 
 
+@wp.func
+def sphere_cylinder(
+  sphere: Geom,
+  cylinder: Geom,
+  worldid: int,
+  d: Data,
+  margin: float,
+  geom_indices: wp.vec2i,
+):
+  """Calculates contact between a sphere and a cylinder."""
+
+  axis = wp.vec3(cylinder.rot[0, 2], cylinder.rot[1, 2], cylinder.rot[2, 2])
+  delta = sphere.pos - cylinder.pos
+
+  # Project and clamp to cylinder length
+  proj = wp.clamp(wp.dot(delta, axis), -cylinder.size[1], cylinder.size[1])
+  closest = cylinder.pos + axis * proj
+
+  radial = sphere.pos - closest
+  radial_len = wp.length(radial)
+
+  # Handle degenerate case when sphere center is on cylinder axis
+  if radial_len < 1e-6:
```

Comment:
nit: where did you get this `1e-6`?  If it's just "any epsilon" please use `types.MJ_MINVAL`